src/content/en/2025/generative-ai.md

```diff
@@ -1,20 +1,413 @@
+
+All statistics in this section refer to the percentage of Google Chrome page loads that utilize these APIs at least once, across all release channels and platforms. It is essential to note that the use of WebAssembly and WebGPU does not confirm AI activity; these are general-purpose APIs frequently utilized for tasks such as complex calculations, 3D visualizations, or gaming.
+
+##### WebAssembly
+
+[WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly) (Wasm) acts as the bytecode for the web. Code written in various programming languages, including C++ and Rust, can be compiled into WebAssembly. It enables developers to write optimized, high-performance code that is executed by the browser's scripting engine on the user's CPU.
+
+<!-- markdownlint-disable-next-line MD051 -->
+WebAssembly has broad browser support, being implemented in all relevant browser engines since 2017. In 2025, the usage of WebAssembly (i.e., the instantiation of a WebAssembly module relative to page loads) experienced 27% linear growth (see [Figure 1](#fig-1)) from being active on 4.44% of page loads in January, and 5.64% in December 2025 (January 2024: 3.37%).
+
+{{ figure_markup(
+  image="genai-wasm-usage.png",
+  caption="WebAssembly usage in 2025 according to Chrome Platform Status data.",
+  description="Bar chart showing the growth of WebAssembly usage in percent per page loads according to Chrome Platform Status data throughout 2025. Usage increased linearly from being active on 4.44% of page loads in January to 5.64% in December.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTXSz19p32lprObXmLblQxEy5u0Sjd9QajNixDEOJutiaxi1aWk37ojoY5Z3D-GfgHg3Ggu23VZS2yI/pubchart?oid=148205691&format=interactive",
+  sheets_gid="303424003"
+  )
+}}
+
+##### WebGPU
+
+[WebGPU](https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API) is the modern successor to [WebGL](https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API), designed to expose the capabilities of modern GPUs to the web. Unlike WebGL, which was strictly for graphics, WebGPU provides support for "compute shaders", allowing for general-purpose computing on graphics cards. This allows developers to perform massive parallel calculations, as required by AI models, directly on the user's graphics card.
+
+<!-- markdownlint-disable-next-line MD051 -->
+WebGPU has become the standard foundation for running AI workloads in the browser. With the release of Firefox 141 in November 2025, [WebGPU became available in all relevant browser engines](https://web.dev/blog/webgpu-supported-major-browsers) (Chromium, Gecko, and WebKit). In 2025, WebGPU experienced exponential growth in page loads, increasing by 147% (see [Figure 2](#fig-2)).
+
+The July 2025 crawl of HTTP Archive data suggested that the API is used on 55,825 desktop sites and 68,043 mobile sites, a significant increase from 4,479 desktop and 4,742 mobile pages from the July 2024 crawl.
```

Comment:
Better to talk in percentages than absolute numbers:
https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide#a-warning-on-totals